### PR TITLE
Feat/add request

### PR DESCRIPTION
**Pull request by Rusli4k**

### CODE REVIEW

File: app/entities/user.go [hunk 1]
```diff
@@ -8,12 +8,12 @@ package entities
...
-	FirstName string
-	LastName  string
-	Password  string
-	CreatedAt string
+	ID        string `json:"id"`
+	Email     string `json:"email"`
+	FirstName string `json:"first-name"`
+	LastName  string `json:"last-name"`
+	Password  string `json:"password"`
+	CreatedAt string `json:"created-at"`
```
Comment (lifehou5e) [inline app/entities/user.go:16]: Do we really need to return created-at? I mean we could use json tag with "-", to not display it through some GET request. IMO it's our internal field.

Comment (delveper) [reply app/entities/user.go:16]: @lifehou5e Maybe we will create some kind of private User without a password and other sensitive fields for some reason in the following iterations?

Comment (lifehou5e) [reply app/entities/user.go:16]: @delveper may be, but we are talking about json tags. Where will they use other than in http response? We don't need it (field "CreatedAt") private, we just don't need to share it through json response.

File: app/transport/http/user.go [hunk 2]
```diff
@@ -11,6 +11,10 @@ func (uh UserHandler) CreateUser() http.Handler {
 	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
 		usr := entities.User{}
 
+		if err := DecodeBody(req, &usr); err != nil {
```
Comment (vpakhuchyi) [inline app/transport/http/user.go:14]: You don't need that function at all. Just put its logic here.
Having it as a separate function doesn't make code more readable.

File: app/entities/user.go (hunk 1 above)
Comment (Rusli4k) [reply app/entities/user.go:16]: This  struct is for all our program, not only for request

Comment (lifehou5e) [reply app/entities/user.go:16]: And where do we need json tags in our program other than in response to http request?

Comment (Rusli4k) [reply app/entities/user.go:16]: i just  get this struct from our DB struck. It must be the same.

Comment (Rusli4k) [reply app/entities/user.go:16]: > And where do we need json tags in our program other than in response to http request?

where have you been, when we tack about it in telegram? )))

Comment (lifehou5e) [reply app/entities/user.go:16]: JSON tags has nothing with sameness, imo)

Comment (Rusli4k) [reply app/entities/user.go:16]: oh, maybe i don`t understand you. Also we have no reason to display password in requests? Its better to make correct export form DB without "created" and "passwrd"...

Comment (lifehou5e) [reply app/entities/user.go:16]: Well, i am waiting for someone who have better knowledge. Cuz right now the only way we interact with json tags for me is through http request/response and if so we don't need to show our internal data. Maybe password too.

File: app/transport/http/user.go (hunk 2 above)
Comment (delveper) [reply app/transport/http/user.go]: @vpakhuchyi  Having helper functions like decode, endode, respErr will make our code more DRY if we are going to propagate our handlers, isnt it?

**Review by lifehou5e — APPROVED** [review]

File: app/transport/http/user.go (hunk 2 above)
Comment (Rusli4k) [reply app/transport/http/user.go]: fix it in new commit

Comment (vpakhuchyi) [reply app/transport/http/user.go]: @delveper you are right
However, there is no sense to move 2 line to a separate function cos these 2 lines may look more simple than a new function with input and output arguments.

Here we tried to encapsulate call to json.Decode().
Just a single line of code or 3 if we count errors handling. 

```
	        if err := json.NewDecoder(req.Body).Decode(&user); err != nil {
			return //someone have to work with errors
		}
		defer req.Body.Close()
```

Is it much easier to read than:
```
                if err := DecodeBody(req, usr); err != nil {
			return
		}
```

I don't think so. Moreover, `DecodeBody` is a custom function, so i have to read it to understand what is going on inside it. But `json.Decode` - is a standard one - i know it.


After all, you have to split code to function when
1. it's reusable (but this isn't) 
2. original function is too big, so you want to simplify it

In the future we will add more rules, but that's all for now.

**Review by vpakhuchyi — APPROVED** [review]

**Review by vdshakh — APPROVED** [review]

**Review by delveper — APPROVED** [review]

**Pull request merged**